server/src/main/scala/za/co/absa/atum/server/api/database/PostgresDatabaseProvider.scala

```diff
@@ -14,16 +14,20 @@
  * limitations under the License.
  */
 
-package za.co.absa.atum.server.api.service
+package za.co.absa.atum.server.api.database
 
-import org.springframework.beans.factory.annotation.Autowired
-import org.springframework.stereotype.Service
-import za.co.absa.atum.server.api.AtumConfig
+import doobie.Transactor
+import za.co.absa.fadb.doobie.DoobieEngine
+import zio._
+import zio.interop.catz._
 
-@Service
-class TestService @Autowired()(atumConfig: AtumConfig){
-  def getMessage: String = {
-    val testConfigVal = atumConfig.testEndpointConfig.getString("some-key")
-    s"The service says: alfa '$testConfigVal'"
+class PostgresDatabaseProvider(val dbEngine: DoobieEngine[Task])
+
+object PostgresDatabaseProvider {
+  val layer: URLayer[Transactor[Task], PostgresDatabaseProvider] = ZLayer {
+    for {
+      transactor <- ZIO.service[Transactor[Task]]
```

Comment:
is `TransactorProvider` somehow gonna be injected here? Because `TransactorProvider` is not used anywhere apart from being imported into `Main.scala` as a layer

Reply:
As discussed on the call. `Transactor[Task]` will be looked up from zio environment. `TransactorProvided` is only an object where the layer is defined.